src/gui/torrentcreatordialog.cpp

```diff
@@ -239,7 +239,7 @@ void TorrentCreatorDialog::handleCreationSuccess(const Path &path, const Path &b
         }
 
         BitTorrent::AddTorrentParams params;
-        params.savePath = branchPath;
+        params.savePath = branchPath + u"/";
```

Comment:
Looks like workaround...
>Fix cannot create torrent when the source file is at root folder.

Shouldn't branchPath point to location of source files? I.e. if "the source file is at root folder" shouldn't it be the root folder?

Reply:
>Shouldn't branchPath point to location of source files? I.e. if "the source file is at root folder" shouldn't it be the root folder?

You can try it by having a root folder at `c:\a` and `c:\a\1.txt` as data. Put path `c:\a` to torrent creator and check "start seeding immediately".
Here `branchPath` will be `c:` but it seems underlying api won't accept it so it is patched to `c:\`.

Reply:
>Shouldn't branchPath point to location of source files? I.e. if "the source file is at root folder" shouldn't it be the root folder?

I'm not sure how `params.savePath` is intended to be used. If you put `c:\a` to `params.savePath` then libtorrent won't find the files.

Reply:
> Here `branchPath` will be `c:` but it seems underlying api won't accept it so it is patched to `c:/`.

I found a real bug. It is `Path::parentPath()` that has special handling of *nix root path but it has no such handling of Windows root paths. Windows root paths is also special case where trailing separator is required. Actually `C:` isn't root folder but something like current folder of disk C. So parent path of `C:/a` should be `C:/`.

P.S. `Path::rootItem()` is also affected by this bug.

Reply:
>I found a real bug. It is Path::parentPath() that has special handling of *nix root path but it has no such handling of Windows root paths. 

I thought about it too but I'm not confident that this is safe to do. Previously the `Path` was under assumption of no trailing `/` character but now it is broken. I'm not sure whether this change will affect other methods correctness...
BTW I often consider qbt should be able to run some unit testing for the auxiliary classes but I'm not sure how to do it without adding another project..

Reply:
> I often consider qbt should be able to run some unit testing for the auxiliary classes

Me too.


> Previously the `Path` was under assumption of no trailing `/` character but now it is broken.

As I said above there is exception for *nix root path. I just missed some "crazy" Windows path cases.
In any case, it should be fixed inside the Path class. If you don't want to do it yourself, then I'll do it later.

Workarounds make the code poorly maintained in the long run, so I'd rather not resort to them.

Reply:
I pushed an update.